File: CodeForces/A/271A.py
def hasDistinct(num):
    while(num > 0):
        digit = num % 10
        remaining = num // 10
        while(remaining > 0):
            if remaining % 10 == digit:
                return False
            remaining //= 10
        num //= 10

    return True

y = int(input())
y += 1
while not hasDistinct(y):
    y += 1


print(y)

File: CodeForces/A/271A.cpp
#include <bits/stdc++.h>
using namespace std; 
typedef long long ll;

bool hasDistinct(int number) {
    if (number < 0) number = -number;

    while(number > 0) {
	int currDigit = number % 10;
	int remaining = number / 10;
	while(remaining > 0) {
	    if(remaining % 10 == currDigit) return false;
	    remaining /=10;
	}
	number /=10;

    }
    return true;
}
int main(void) {
    int y;
    cin >> y;

    do {
        y++;
    } while (!hasDistinct(y));

    cout << y << endl;
    return 0;
}
